### ISSUE TITLE
Feature request: Add target or _target option to URL or redirect

### ISSUE THREAD

**Issue by jcrmatos** [web2py/web2py#2141]
Hello,

I have this function

```
def check_only_one_selection(ids, destination):
    # type: (List[int], str) -> None
    """Check only one selection in grid.

    :param ids: List of Ids.
    :param destination: Destination page if one selection was made.
    """
    if len(ids) > 1:
        session.flash = T('Select only one record.')
    else:
        redirect(URL(destination, args=[ids[0]])
```

and would like it to open it's view on another web page (not the same from where it was called).
Something similar to this
`<a target="_blank">`
but without using A.

I would like to suggest adding the target or _target option to URL or redirect.

Thanks,

JM

**Comment by timnyborg** [comment]
I don't think this is possible.  Unless I'm mistaken, a redirect (301 or otherwise) cannot specify a new tab.

This sort of thing would require some AJAX.  Call the function via AJAX, check if the response is a URL, and open a new tab accordingly.

**Comment by leonelcamara** [comment]
It's sort of possible in the redirect(..., client_side=True) case, but browsers will consider this a pop-up and block it.

**Comment by jcrmatos** [comment]
@leonelcamara I tried your solution, but the page opens in the same tab. Do I have to do anything else besides adding client_side=True to redirect?
I have this
`redirect(URL(destination, args=[ids[0]],  user_signature=True), client_side=True)`

**Comment by leonelcamara** [comment]
No that should do a redirect, note that as it's implemented right now it doesn't open in a new window, you can edit your application's web2py.js to make it do so.

**Comment by jcrmatos** [comment]
Thanks @leonelcamara but I have no idea how to do that.

**Comment by leonelcamara** [comment]
@jcrmatos look at this line

https://github.com/web2py/web2py/blob/master/applications/welcome/static/js/web2py.js#L283

In that handler instead of doing window.location = redirect, you can do a window.open(redirect, '_blank')

**Comment by jcrmatos** [comment]
Hello,

Thanks, but I solved it with this "insane" solution.
In the function responsible for the original redirect I changed it to itself with the grid selection as a request.var.

`        redirect(URL(vars={'pip': ids[0]}, user_signature=True))`

Then on the original page I added this code at the end of the page

```
{{if request.vars and request.vars.pip:}}
<script><!--
    window.onload = function redirect_to_new_tab() {
        window.open("{{=URL('open_wo', 'print_id_page', args=[request.vars.pip], user_signature=True)}}", "_blank");
        window.location.href = "{{=URL('open_wo', 'index', user_signature=True)}}";
    }
//--></script>
{{pass}}
```

This worked because the user sees the original page and the new page to print out. I have to allow popups, but that is ok.

But it would still be nice to have the option target or _target directly in the redirect. :)

Best regards,

JM
